package.json

```diff
@@ -165,6 +165,7 @@
     ]
   },
   "dependencies": {
+    "@ethersproject/hash": "^5.5.0",
```

Comment:
The algorithm for hashing names is very simple, I think we can implement it ourselves without bringing an external dependency https://docs.ens.domains/contract-api-reference/name-processing#hashing-names

Reply:
https://github.com/ethers-io/ethers.js/blob/4e6d121fb8aa7327290afab7653364be8ddd8d81/packages/hash/src.ts/namehash.ts#L27-L47

Reply:
Should it change or be adjusted we wouldn't realise though?

I want to propose the migration from web3 to ethers anyway due to bundle size, the latter has a built in function for reverse lookup.

Reply:
> Should it change or be adjusted we wouldn't realise though?

I don't think they can change such things, they may release a new standard or similar but not change the existing implementation as it would break a lot of things

Reply:
The function you linked calls other functions that we do not have access to though?

Reply:
Ok, we'll come back to this later on today.

Reply:
You can check the implementation of web3.eth.accounts.hashMessage to check what functions web3.js uses under the hood

Reply:
It looks like web3.js currently doesn't provide all the functions that we need to replace `namehash` with our own implementation, i.e. `nameprep`. I would suggest that we keep using the hash library from ethers for now or handle the implementation for `namehash` in a separate issue since there are other PRs that are dependant on the ENS reverse lookup implemented in this PR.